test/e2e/gateway/http_route_partially_conflict_match_test.go

```diff
@@ -0,0 +1,105 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayapi_v1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	"github.com/projectcontour/contour/internal/gatewayapi"
+	"github.com/projectcontour/contour/test/e2e"
+)
+
+func testHTTPRoutePartiallyConflictMatch(namespace string, gateway types.NamespacedName) {
+	Specify("Creates two http routes, second one has partial conflict match against the first one, has partially match condition", func() {
+		By("create httproute-1 first")
+		route1 := &gatewayapi_v1.HTTPRoute{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Namespace: namespace,
+				Name:      "httproute-1",
+			},
+			Spec: gatewayapi_v1.HTTPRouteSpec{
+				Hostnames: []gatewayapi_v1.Hostname{"queryparams.gateway.projectcontour.io"},
+				CommonRouteSpec: gatewayapi_v1.CommonRouteSpec{
+					ParentRefs: []gatewayapi_v1.ParentReference{
+						gatewayapi.GatewayParentRef(gateway.Namespace, gateway.Name),
+					},
+				},
+				Rules: []gatewayapi_v1.HTTPRouteRule{
+					{
+						Matches: []gatewayapi_v1.HTTPRouteMatch{
+							{QueryParams: gatewayapi.HTTPQueryParamMatches(map[string]string{"animal": "whale"})},
+						},
+						BackendRefs: gatewayapi.HTTPBackendRef("echo-1", 80, 1),
+					},
+					{
+						Matches: []gatewayapi_v1.HTTPRouteMatch{
+							{QueryParams: gatewayapi.HTTPQueryParamMatches(map[string]string{"animal": "dolphin"})},
+						},
+						BackendRefs: gatewayapi.HTTPBackendRef("echo-2", 80, 1),
+					},
+				},
+			},
+		}
+		f.CreateHTTPRouteAndWaitFor(route1, e2e.HTTPRouteAccepted)
+
+		By("create httproute-2 with only partially conflicted matches")
+		route2 := &gatewayapi_v1.HTTPRoute{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Namespace: namespace,
+				Name:      "httproute-2",
+			},
+			Spec: gatewayapi_v1.HTTPRouteSpec{
+				Hostnames: []gatewayapi_v1.Hostname{"queryparams.gateway.projectcontour.io"},
+				CommonRouteSpec: gatewayapi_v1.CommonRouteSpec{
+					ParentRefs: []gatewayapi_v1.ParentReference{
+						gatewayapi.GatewayParentRef(gateway.Namespace, gateway.Name),
+					},
+				},
+				Rules: []gatewayapi_v1.HTTPRouteRule{
+					{
+						Matches: []gatewayapi_v1.HTTPRouteMatch{
+							{QueryParams: gatewayapi.HTTPQueryParamMatches(map[string]string{"animal": "whale"})},
+						},
+						BackendRefs: gatewayapi.HTTPBackendRef("echo-1", 80, 1),
+					},
+					{
+						Matches: []gatewayapi_v1.HTTPRouteMatch{
+							{QueryParams: gatewayapi.HTTPQueryParamMatches(map[string]string{"no-conflict": "no-joke", "no-conflict-again": "no-joke"})},
+						},
+						BackendRefs: gatewayapi.HTTPBackendRef("echo-2", 80, 1),
+					},
+				},
+			},
+		}
+		// Partially accepted
+		f.CreateHTTPRouteAndWaitFor(route2, e2e.HTTPRoutePartiallyAccepted)
+		// Still has Accepted: true
+		require.Eventually(f.T(), func() bool {
+			if err := f.Client.Get(context.TODO(), client.ObjectKeyFromObject(route2), route2); err != nil {
+				return false
+			}
+			return e2e.HTTPRouteAccepted(route2)
+		}, time.Minute*2, f.RetryInterval)
```

Comment:
I think you can combine these:
```suggestion
		f.CreateHTTPRouteAndWaitFor(route2, func(...) {
		        return e2e.HTTPRoutePartiallyAccepted(route2) && e2e.HTTPRouteAccepted(route2)
		})
```